clusterloader2/pkg/measurement/common/kube_state_metrics_measurement.go

```diff
@@ -0,0 +1,201 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"k8s.io/perf-tests/clusterloader2/pkg/measurement"
+	measurementutil "k8s.io/perf-tests/clusterloader2/pkg/measurement/util"
+	"k8s.io/perf-tests/clusterloader2/pkg/util"
+
+	"github.com/prometheus/common/model"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/klog"
+)
+
+const (
+	ksmLatencyName               = "KubeStateMetricsLatency"
+	ksmRequestDurationMetricName = model.LabelValue("http_request_duration_seconds_bucket")
+	probeIntervalDefault         = 30 * time.Second
+)
+
+type ksmLatencyMeasurement struct {
+	ctx         context.Context
+	cancel      func()
+	isRunning   bool
+	stopCh      chan struct{}
+	namespace   string
+	serviceName string
+	metricsPort int
+	selfPort    int
+	latency     *measurementutil.Histogram
+	wg          sync.WaitGroup
+}
+
+func init() {
+	if err := measurement.Register(ksmLatencyName, CreateKSMLatencyMeasurement); err != nil {
+		klog.Fatalf("Cannot register %s: %v", ksmLatencyName, err)
+	}
+}
+
+// CreateKSMLatencyMeasurement creates a new Kube State
+// Metrics Measurement.
+func CreateKSMLatencyMeasurement() measurement.Measurement {
+	return &ksmLatencyMeasurement{
+		namespace:   "kube-state-metrics-perf-test",
+		serviceName: "kube-state-metrics",
+		selfPort:    8081,
+		metricsPort: 8080,
+	}
+}
+
+// Execute supports two actions:
+// - start - starts goroutine and queries /metrics every probeIntervalDefault interval,
+// it also collects initial latency metrics.
+// - gather - gathers latency metrics and creates a latency summary.
+func (m *ksmLatencyMeasurement) Execute(config *measurement.Config) ([]measurement.Summary, error) {
+	if m.ctx == nil {
+		m.ctx, m.cancel = context.WithCancel(context.Background())
```

Comment:
There is no way to pass it here.

So given that, please simply initialize this in CreateKSMLatencyMeasurement